src/pages/index.js

```diff
@@ -6,87 +6,230 @@ import {
   selectors,
   editFormEl,
   addFormEl,
+  pictureFormEl,
   validationSettings,
   profileEditBtn,
   cardAddBtn,
   profileInfo,
   cardAddForm,
   profileNameInput,
   profileDescriptionInput,
+  pictureEditBtn,
+  pictureLinkInput,
+  pictureEl,
+  cardLikeBtn,
+  cardDeleteBtn,
+  confirmModalBtn,
 } from "../utils/constants";
 import Card from "../components/Card";
 import FormValidator from "../components/FormValidator";
 import Section from "../components/Section";
 import PopupWithImage from "../components/PopupwithImage";
 import PopupwithForm from "../components/PopupwithForm";
+import PopupWithConfirm from "../components/PopupWithConfirm";
 import UserInfo from "../components/UserInfo";
+import Api from "../components/Api";
+
+let cardSection;
+
+// API //
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "40adfe4e-c0e4-40a1-9506-b555db893f72",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          const initialCard = createCard(cardData);
+          cardSection.addItem(initialCard);
+        },
+      },
+      ".gallery__cards"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => console.error(`Error setting up card section, ${err}`));
+
+api
+  .getUserInfo()
+  .then((data) => {
+    profileInfoClass.setUserInfo({
+      name: data.name,
+      about: data.about,
+      avatar: data.avatar,
+    });
+  })
+  .catch((err) => console.error(`Error loading profile info. ${err}`));
+
+const createCard = (cardData) => {
+  const card = new Card(
+    {
+      name: cardData.name,
+      link: cardData.link,
+      isLiked: cardData.isLiked,
+      _id: cardData._id,
+    },
+    "#card-template",
+    handleImageClick,
+    handleDeleteBtn,
+    handleLikeBtn
+  );
+  return card.getView();
+};
 
 // CLASSES //
 
 const profileInfoClass = new UserInfo(profileInfo);
 
+const pictureFormClass = new PopupwithForm(
+  "#profile-picture-modal",
+  handlePictureSubmit
+);
+
 const profileFormClass = new PopupwithForm(
   "#profile-edit-modal",
   handleProfileFormSubmit
 );
+
 const cardAddFormClass = new PopupwithForm(
   "#add-card-modal",
   handleCardAddSubmit
 );
+
+const confirmFormClass = new PopupWithConfirm("#confirm-modal");
+
 const cardPreviewClass = new PopupWithImage("#image-modal");
-const cardSectionClass = new Section(
-  {
-    items: cardData,
-    renderer: renderCard,
-  },
-  selectors.cardSection
-);
+
 const editFormValidator = new FormValidator(validationSettings, editFormEl);
 const addFormValidator = new FormValidator(validationSettings, addFormEl);
+const pictureFormValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```